src/platform/Darwin/DnssdContexts.cpp

```diff
@@ -536,7 +526,17 @@ void ResolveContext::DispatchSuccess()
 
     for (auto interfaceIndex : priorityInterfaceIndices)
     {
-        if (TryReportingResultsForInterfaceIndex(static_cast<uint32_t>(interfaceIndex)))
+        // Try finding interfaces for domains kLocalDot and kOpenThreadDot and delete them.
```

Comment:
This comment should say something like:

> Prioritize results on the interfaces in priorityInterfaceIndices for kLocalDot and kOpenThreadDot domains.

But this raises the question: why are we not just prioritizing them for _all domains_?

Reply:
The problem here is I do not know what all domains to check for. For now we have the list of priority interfaces that have index ids but no domain but the interface key is a combination of interface ids + domain name. I am not sure if i can get all entries for an interface id from the map (can i provide a wild card domain name or something like that)?

Reply:
if you look in the latter part of the code [here](https://github.com/project-chip/connectedhomeip/pull/32631/files#diff-d3d1d99d691d0aef90e9de721cad391450d0e28aa0b727d812022acc3755f248R551) we are prioritizing whatever's in the map. for the deletion part since we have a harcoded list of interface indices, i am not sure what to use for the domain? if i could get all interfaces from the map that have the interface index I could iterate through them check if they need deletion and delete them

Reply:
You should not be validating the domain. If you get a response, the response is valid. It's completely permissible and correct for a resolve, for example, to return a hostname in the SRV record that's looked up by DNS rather than mDNS, e.g. www.apple.com.

DNS responses always have an interface index of 0, and this includes lookups in the default.service.arpa domain. (We have a bug with this right now, but that's the intended behavior).

Reply:
I added the comment to indicate that I am deleting the interfaces with those 2 domains. AFAIK the code at line 551 is prioritizing all domains.

Reply:
fixed this by iterating through the map. If the interface index matches the key, we prioritize the results.